pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -845,6 +845,15 @@ func (r *Builder) mapDisks(vm *model.VM, vmRef ref.Ref, persistentVolumeClaims [
 				// fallback to default name and reset error
 				volumeName = fmt.Sprintf("vol-%v", i)
 			}
+
+			// check if the generated name is longer then 63 characters
+			if len(volumeName) > 63 {
```

Comment:
Can we know this information during plan validations? So we would tell it to the user so they change the template

Reply:
hmm, it may depend on the PVC name, that may depend on the vm name ...
a lot of options, for example
is the vm name - the vm name, automatically change, have a vm anem template ?
then
is the PVC name ...
then
what is the volume name, and is it longer then 64 ....

Reply:
hmm okay it's an edge case, lets document this so the users will be aware of the possibility of truncating the PVC name